src/App.ts

```diff
@@ -584,24 +604,41 @@ export default class App {
    * via {@link String.prototype.includes}.
    * @param listeners Middlewares that process and react to the message events that matched the provided pattern.
    */
-  public message(
-    filter: MessageEventMiddleware, pattern: string | RegExp, ...listeners: MessageEventMiddleware[]
+  public message<
+      MiddlewareCustomContext extends StringIndexed = StringIndexed,
+  >(
+    filter: MessageEventMiddleware<AppCustomContext & MiddlewareCustomContext>,
+    pattern: string | RegExp,
+    ...listeners: MessageEventMiddleware<AppCustomContext & MiddlewareCustomContext>[]
   ): void;
   /**
    *
    * @param filter Middleware that can filter out messages. Generally this is done by returning before calling
    * {@link AllMiddlewareArgs.next} if there is no match. See {@link directMention} for an example.
    * @param listeners Middlewares that process and react to the message events that matched the provided patterns.
    */
-  public message(filter: MessageEventMiddleware, ...listeners: MessageEventMiddleware[]): void;
+  public message<
+    MiddlewareCustomContext extends StringIndexed = StringIndexed,
+  >(
+    filter: MessageEventMiddleware,
+    ...listeners: MessageEventMiddleware<AppCustomContext & MiddlewareCustomContext>[]
+  ): void;
```

Comment:
While building the tests I wasn't able to get TypeScript to use this overload. As the filter and ...listeners share the same type I'm not sure this overload is necessary. If there's agreement, I feel like the first overloads docs could be updated to go into more detail about how the filters can work.
- https://github.com/slackapi/bolt-js/blob/7eec4389253ad5b353af8d3cf0a15552eb18f4f1/src/App.ts#L567-L571

Reply:
Good catch. As long as it still works without the overloaded method, I think that we can safely remove the unnecessary method with `filter: MessageEventMiddleware` argument.